code/ui/components/src/Zoom/browserSupportsCssZoom.ts

```diff
@@ -1,10 +1,12 @@
 export function browserSupportsCssZoom(): boolean {
   try {
-    // Checks if safari or firefox is being used.
-    // This check can be removed when zoom becomes standard css. Currently firefox does not support it and there is a bug in safari see here:  https://developer.mozilla.org/en-US/docs/Web/CSS/zoom#browser_compatibility
-    // regex checks if there are other browsers because "safari" is also present when using chrome or an android browser whilst on mac/iPhone see more here : https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent#which_part_of_the_user_agent_contains_the_information_you_are_looking_for
-    const isCompatible = /^((?!chrome|android).)*safari|firefox/i.test(navigator.userAgent);
-    return isCompatible;
+    // First checks if Safari is being used, because Safari supports zoom, but it's buggy: https://developer.mozilla.org/en-US/docs/Web/CSS/zoom#browser_compatibility
+    if (/safari/i.test(navigator.userAgent)) {
```

Comment:
Chrome returns `true` in here on a Mac

`/safari/i.test(navigator.userAgent)`

Chrome Canary: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36'
Safari: "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.3 Safari/605.1.15" = $1

It causes the fallback mechanics to kick in on Both chroma nad safari and story Which I had before (the actual component is blurred because it's irrelevant what it is):

<img width="728" alt="Screenshot 2023-02-20 at 14 43 29" src="https://user-images.githubusercontent.com/948356/220124779-711b8b58-5c07-4ff8-8960-3db51904b359.png">

becomes:

<img width="724" alt="Screenshot 2023-02-20 at 14 44 12" src="https://user-images.githubusercontent.com/948356/220124853-40c8055d-4e40-416f-a000-03fbd6dde263.png">


(Some bug with how `ZoomElementWrapper` works surfaces)

The story Which is shown above is displayed on MDX document like this:

```mdx
import * as whateverStories from './whatever.stories';

<Meta of={ whateverStories } />

<Stories includePrimary={ true }></Stories>
```

Reply:
Thanks, I think you're right, I'm not sure how this slipped past us since we're all using Chrome on Mac. I'll fix it.

Reply:
https://github.com/storybookjs/storybook/pull/21163

Reply:
Right, that does the job (at least it won't apply the fallback in Chrome)

It looks, though, as if there's actually a problem with capturing height of the story. And Safari will still be affected.
it does not block me personally, but it seems like it will be a problem.

I don't know how to properly narrow it down - in which part of Storybook to look for the root cause.
The only Storybook I maintain is the one using `@storybook/web-components-vite` so it's quite a niche setup.

I have a feeling that the `20px` that's assigned as height is measured before my custom element is loaded and 20 is result of sum of top and bottom border of the `.innerZoomElementWrapper>*`:

<img width="725" alt="Screenshot 2023-02-20 at 16 09 31" src="https://user-images.githubusercontent.com/948356/220142718-0b1912cc-935e-4348-b11e-6c8fe53402f8.png">

That's not a problem by itself - the problem seems to be that after the custom element loads the height is not adjusted, where it seems like it should:

https://github.com/storybookjs/storybook/blob/446c3236e0f4edcaa426e6f8225ef096379daf2e/code/ui/components/src/Zoom/ZoomElement.tsx#L74

https://github.com/storybookjs/storybook/blob/446c3236e0f4edcaa426e6f8225ef096379daf2e/code/ui/components/src/Zoom/ZoomElement.tsx#L61-L63

If I understand correctly what I read in

https://github.com/storybookjs/storybook/blob/446c3236e0f4edcaa426e6f8225ef096379daf2e/code/ui/components/src/Zoom/ZoomElement.tsx#L24-L48

then it may be right - there are no mutations in my stories, because markup for custom element does not change between a moment where story gets rendered initially and when component is loaded

Reply:
Hmm that is interesting. So just to understand you correctly, in Safari (and Firefox), the Canvas height isn't responsive to the contained component, it's always the same height as the initial load? And even if you don't zoom, it's still too small?

I think it would be good if you could create a separate bug report for this with a minimal reproduction. I don't think that issue is strictly related to this PR, as this just fixed the browser detection, and it sounds like your issue is more fundamental, regardless of correct or bad detection.

Reply:
@JReinhold I think https://github.com/storybookjs/storybook/pull/21174 will be a solution